src/render/filters.rs

```diff
@@ -234,6 +235,70 @@ impl ResolveFilter for ExternalFilter {
     }
 }
 
+impl ResolveFilter for FirstFilter {
+    fn resolve<'t, 'py>(
+        &self,
+        variable: Option<Content<'t, 'py>>,
+        py: Python<'py>,
+        _template: TemplateString<'t>,
+        _context: &mut Context,
+    ) -> ResolveResult<'t, 'py> {
+        let content = match variable {
+            Some(content) => {
+                // Try to get the first item from the content
+                match content {
```

Comment:
We can save a lot of horizontal space by moving the `match content` to top level:

```rust
        let content = match variable {
            Some(content) => content,
            None => return Ok("".as_content()),
        };
        let content = match content {
```